torch/csrc/autograd/engine.cpp

```diff
@@ -793,14 +798,16 @@ void Engine::mark_graph_task_completed(std::shared_ptr<GraphTask>& graph_task) {
   }
 
   try {
-    auto val = graph_task_exec_post_processing(graph_task);
-    graph_task->future_result_->markCompleted(val);
+    // Run post processing, before marking the future as complete.
+    graph_task_exec_post_processing(graph_task);
+    graph_task->future_result_->markCompleted(
+        std::move(graph_task->captured_vars_));
```

Comment:
It would be good to specify in the docs of `captured_vars_` that it is moved out of (and thus no longer valid on the graph task) after post processing.